lib/concurrent/synchronization/immutable_struct.rb

```diff
@@ -0,0 +1,67 @@
+module Concurrent
+  module Synchronization
+    # Similar to Struct but the fields are immutable and always visible (like Java final fields)
+    # @example
+    #   Person = ImmutableStruct.with_fields :name, :age
+    #   Person.new 'John Doe', 15
+    #   Person['John Doe', 15]
+    #   Person['John Doe', 15].members # => [:name, :age]
+    #   Person['John Doe', 15].values  # => ['John Doe', 15]
+    class ImmutableStruct < Synchronization::Object
+      def self.with_fields(*names, &block)
```

Comment:
Yeah you are right I should have made it to match in the first place. I'll appreciate your help, I am not sure how to deal with the `new` method, where it in one case constructs classes and in the second instances.